extension/content/components/recipes/form/arguments/BranchedAddon.js

```diff
@@ -0,0 +1,218 @@
+// @ts-nocheck
+
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Col,
+  ControlLabel,
+  Divider,
+  FormGroup,
+  HelpBlock,
+  Icon,
+  IconButton,
+  Input,
+  InputNumber,
+  Panel,
+  Row,
+  SelectPicker,
+} from "rsuite";
+
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+import {
+  useEnvironmentState,
+  useSelectedNormandyEnvironmentAPI,
+} from "devtools/contexts/environment";
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+
+export default function BranchedAddon() {
+  const { selectedKey: environmentKey } = useEnvironmentState();
+  const [extensions, setExtensions] = React.useState([]);
+  const normandyApi = useSelectedNormandyEnvironmentAPI();
+
+  React.useEffect(() => {
+    normandyApi.fetchAllExtensions().then((allExtensions) => {
+      setExtensions(allExtensions);
+    });
+  }, [environmentKey]);
+
+  const options = extensions.map((extension) => ({
+    label: extension.name,
+    value: extension.id,
+  }));
+
+  return (
+    <FormGroup>
+      <Row>
+        <Col xs={12}>
+          <InputField label="Slug" name="slug" />
+          <InputField
+            label="User Facing Description"
+            name="userFacingDescription"
+          />
+        </Col>
+        <Col xs={12}>
+          <InputField label="User Facing Name" name="userFacingName" />
+          <ToggleField label="Prevent New Enrollment" name="isEnrollmentPaused">
+            Prevents new users from joining this study cohort. Exisiting users
+            will remain in the study.
+          </ToggleField>
+        </Col>
+      </Row>
+      <Branches options={options} />
+    </FormGroup>
+  );
+}
+
+function Branches({ options }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleClickAddBranch = () => {
+    /** @type {{ratio: number, slug: string, value: string | number | boolean}} */
```

Comment:
you have `// @ts-nocheck` above. is that intended?

Reply:
errm yes? cause typescript type checking was complaining about other things too?